http/unstable_route.ts

```diff
@@ -96,15 +169,32 @@ export function route(
   return (request: Request, info?: Deno.ServeHandlerInfo) => {
     for (const route of routes) {
       const match = route.pattern.exec(request.url);
-      if (
-        match &&
-        (Array.isArray(route.method)
-          ? route.method.includes(request.method)
-          : request.method === (route.method ?? "GET"))
-      ) {
-        return route.handler(request, match, info);
+      if (match === null) {
+        continue;
+      }
+
+      let handler: Handler;
+      if (route.handler) {
+        handler = route.handler;
+      } else if (!(request.method in route.handlers)) {
+        /**
+         * @see {@link https://www.iana.org/go/rfc2616 | RFC2616, Section 14.7}
+         */
+        return new Response("Method Not Allowed", {
+          status: 405,
+          headers: { Allow: Object.keys(route.handlers).join(", ") },
+        });
+      } else {
+        handler = route.handlers[request.method] as Handler;
+      }
+
+      try {
+        return handler(request, match, info);
+      } catch (error) {
+        return errorHandler(error, request, match, info);
       }
     }
-    return defaultHandler(request, info);
+
+    return new Response("Not Found", { status: 404 });
```

Comment:
I don't think it's a good idea to remove the capability to customize the 404 page